### PR TITLE
Finish HOME UI

### PR DESCRIPTION
**Pull request by cungtai**

### CODE REVIEW

File: app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.fstyle.androidtrainning.screen.favourite;
+
+import android.support.v4.app.Fragment;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java]: ERROR: Unused import: android.support.annotation.Nullable. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.fstyle.androidtrainning.R;
+
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class FavouriteActivity extends Fragment {
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/HomeActivity.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import android.view.ViewGroup;
+import com.fstyle.androidtrainning.R;
+
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class HomeActivity extends Fragment {
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/HomeActivity.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        setContentView(R.layout.activity_home);
+        mtabLayout = (TabLayout) findViewById(R.id.tabLayout);
+        mtabLayout.addTab(mtabLayout.newTab().setText("Home"));
+        mtabLayout.addTab(mtabLayout.newTab().setText("Search"));
+        mtabLayout.addTab(mtabLayout.newTab().setText("Favorite"));
+        mtabLayout.addTab(mtabLayout.newTab().setText("Profile"));
+        mtabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
+        setupTabIcons();
+        mviewPager = (ViewPager) findViewById(R.id.pager);
+        PagerAdapter adapter = new PagerAdapter(getSupportFragmentManager(), mtabLayout.getTabCount());
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * Created by MyPC on 20/12/2017.
+ */
+public class PagerAdapter extends FragmentStatePagerAdapter {
+//    public PagerAdapter(FragmentManager fragmentManager){
+//        super(fragmentManager);
+//    }
+int mNumOfTabs;
+
+
+    public PagerAdapter(FragmentManager fm, int NumOfTabs) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java]: ERROR: Name 'NumOfTabs' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/fstyle/androidtrainning/screen/profile/ProfileActivity.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import com.fstyle.androidtrainning.R;
+
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class ProfileActivity extends Fragment {
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/profile/ProfileActivity.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/fstyle/androidtrainning/screen/search/SearchActivity.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import android.view.ViewGroup;
+import com.fstyle.androidtrainning.R;
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class SearchActivity extends Fragment{
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/search/SearchActivity.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java (hunk 1 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.support.annotation.Nullable'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/build.gradle [hunk 8]
```diff
@@ -28,11 +28,12 @@ android {
...
-  compile 'com.android.support.constraint:constraint-layout:1.0.2'
-  testCompile 'junit:junit:4.12'
+    compile fileTree(include: ['*.jar'], dir: 'libs')
+    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
+    compile 'com.android.support:appcompat-v7:26.1.0'
+    compile 'com.android.support.constraint:constraint-layout:1.0.2'
+    testCompile 'junit:junit:4.12'
+    implementation 'com.android.support:design:27.0.2'
```
Comment (trainer-android-fr) [inline app/build.gradle]: FATAL: [GradleCompatible] This support library should not use a different version (27) than the `compileSdkVersion` (26)
 There are some combinations of libraries, or tools and libraries, that are incompatible, or can lead to bugs. One such incompatibility is compiling with a version of the Android support libraries that is not the latest version (or in particular, a version lower than your `targetSdkVersion`.)

File: app/src/main/res/layout/activity_favourite.xml [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:text="favorite"
```
Comment (trainer-android-fr) [inline app/src/main/res/layout/activity_favourite.xml]: WARNING: [HardcodedText] Hardcoded string "favorite", should use `@string` resource
 Hardcoding text attributes directly in layout files is bad for several reasons:

* When creating configuration variations (for example for landscape or portrait)you have to repeat the actual text (and keep it up to date when making changes)

* The application cannot be translated to other languages by just adding new translations for existing string resources.

There are quickfixes to automatically extract this hardcoded string into a resource lookup.

File: app/src/main/res/layout/activity_profile.xml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:text="Profile"
```
Comment (trainer-android-fr) [inline app/src/main/res/layout/activity_profile.xml]: WARNING: [HardcodedText] Hardcoded string "Profile", should use `@string` resource
 Hardcoding text attributes directly in layout files is bad for several reasons:

* When creating configuration variations (for example for landscape or portrait)you have to repeat the actual text (and keep it up to date when making changes)

* The application cannot be translated to other languages by just adding new translations for existing string resources.

There are quickfixes to automatically extract this hardcoded string into a resource lookup.

File: app/src/main/res/layout/activity_search.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:text="search"
```
Comment (trainer-android-fr) [inline app/src/main/res/layout/activity_search.xml]: WARNING: [HardcodedText] Hardcoded string "search", should use `@string` resource
 Hardcoding text attributes directly in layout files is bad for several reasons:

* When creating configuration variations (for example for landscape or portrait)you have to repeat the actual text (and keep it up to date when making changes)

* The application cannot be translated to other languages by just adding new translations for existing string resources.

There are quickfixes to automatically extract this hardcoded string into a resource lookup.

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+        setContentView(R.layout.activity_home);
+        mtabLayout = (TabLayout) findViewById(R.id.tabLayout);
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_home));
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_search));
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_favourite));
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_profile));
+        mtabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
+        setupTabIcons();
+        mviewPager = (ViewPager) findViewById(R.id.pager);
+        PagerAdapter adapter = new PagerAdapter(getSupportFragmentManager(), mtabLayout.getTabCount());
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class FavouriteActivity extends Fragment {
+
+    @Override
+    public View onCreateView(LayoutInflater inflater
+            , ViewGroup container,Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java]: ERROR: ',' should be on the previous line.

Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/HomeActivity.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+import com.fstyle.androidtrainning.R;
+
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class HomeActivity extends Fragment {
+    @Override
+    public View onCreateView(LayoutInflater inflater
+            , ViewGroup container, Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/HomeActivity.java]: ERROR: ',' should be on the previous line.

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 15]
```diff
@@ -0,0 +1,61 @@
+package com.fstyle.androidtrainning.screen.home;
+
+import android.os.Bundle;
+import android.support.design.widget.TabLayout;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TableLayout;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: ERROR: Unused import: android.widget.TableLayout. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+        mtabLayout = (TabLayout) findViewById(R.id.tabLayout);
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_home));
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_search));
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_favourite));
+        mtabLayout.addTab(mtabLayout.newTab().setText(R.string.title_tab_profile));
+        mtabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
+        setupTabIcons();
+        mviewPager = (ViewPager) findViewById(R.id.pager);
+        PagerAdapter adapter = new PagerAdapter(getSupportFragmentManager()
+                , mtabLayout.getTabCount());
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: ERROR: ',' should be on the previous line.

File: app/src/main/java/com/fstyle/androidtrainning/screen/profile/ProfileActivity.java [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class ProfileActivity extends Fragment {
+
+    @Override
+    public View onCreateView(LayoutInflater inflater
+            , ViewGroup container, Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/profile/ProfileActivity.java]: ERROR: ',' should be on the previous line.

File: app/src/main/java/com/fstyle/androidtrainning/screen/search/SearchActivity.java [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+import com.fstyle.androidtrainning.R;
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class SearchActivity extends Fragment{
+
+    @Override
+    public View onCreateView(LayoutInflater inflater
+            , ViewGroup container, Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/search/SearchActivity.java]: ERROR: ',' should be on the previous line.

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java (hunk 15 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.widget.TableLayout'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/build.gradle [hunk 19]
```diff
@@ -28,11 +28,11 @@ android {
...
-  compile 'com.android.support:appcompat-v7:26.1.0'
-  compile 'com.android.support.constraint:constraint-layout:1.0.2'
-  testCompile 'junit:junit:4.12'
+    compile fileTree(include: ['*.jar'], dir: 'libs')
+    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
+    compile 'com.android.support.constraint:constraint-layout:1.0.2'
+    testCompile 'junit:junit:4.12'
+    compile 'com.android.support:design:26.0.0'
```
Comment (trainer-android-fr) [inline app/build.gradle]: WARNING: [GradleDependency] A newer version of com.android.support:design than 26.0.0 is available: 26.1.0
 This detector looks for usages of libraries where the version you are using is not the current stable release. Using older versions is fine, and there are cases where you deliberately want to stick with an older version. However, you may simply not be aware that a more recent version is available, and that is what this lint check helps find.

File: app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Created by ossierra on 12/27/17.
+ */
+
+public class FavouriteActivity extends Fragment {
+
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+            ViewGroup container,Bundle savedInstanceState) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/favourite/FavouriteActivity.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+
+import android.os.Bundle;
+import android.support.design.widget.TabLayout;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import com.fstyle.androidtrainning.R;
+
+public class MainActivity extends AppCompatActivity{
+
+    private TabLayout mtabLayout;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: pls rename ``mtabLayout`` -> ``mTabLayout``

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+    private ViewPager mviewPager;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: Pls rename ``mviewPager`` -> ``mViewPager``

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+            @Override
+            public void onTabUnselected(TabLayout.Tab tab) {
+
+            }
+
+            @Override
+            public void onTabReselected(TabLayout.Tab tab) {
+
+            }
+        });
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: Tất cả code trong method ``onCreate`` này nên chia ra 2 method nhỏ hơn
- 1 method với tên ``initViews`` khai báo view kiểu như mTabLayout = findviewbyId(R.id. tabLayout)
- 1 method với tên ``handleEvents`` chứa code phần còn lại

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+import com.fstyle.androidtrainning.screen.search.SearchActivity;
+
+
+/**
+ * Created by MyPC on 20/12/2017.
+ */
+public class PagerAdapter extends FragmentStatePagerAdapter {
+//    public PagerAdapter(FragmentManager fragmentManager){
+//        super(fragmentManager);
+//    }
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java]: xóa comment code

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+int mNumOfTabs;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java]: 1. formart code đoạn này anh ạ
2. đổi tên ``mNumOfTabs`` -> ``mNumOfTab``

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+
+    public PagerAdapter(FragmentManager fm, int numOfTabs) {
+        super(fm);
+        this.mNumOfTabs = numOfTabs;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        switch (position){
+            case 0:
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java]: nên khai báo 0 là 1 biến hằng ở trên cùng của file
```java
private static final int TAB_HOME = 0;
```

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+                HomeActivity frmHome = new HomeActivity();
+                return frmHome;
+            case 1:
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java]: như trên

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+                SearchActivity frmSearch = new SearchActivity();
+                return frmSearch;
+            case 2:
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java]: như trên

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+                FavouriteActivity frmFavourite = new FavouriteActivity();
+                return frmFavourite;
+            case 3:
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/PagerAdapter.java]: như trên

File: app/src/main/res/layout/activity_search.xml [hunk 30]
```diff
@@ -0,0 +1,15 @@
...
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:text="@string/empty_data"
+        android:textAppearance="?android:attr/textAppearanceLarge"/>
+
+</RelativeLayout>
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_search.xml]: cuối file nên thêm 1 dòng trống

File: app/src/main/res/layout/activity_profile.xml [hunk 31]
```diff
@@ -0,0 +1,15 @@
...
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:text="@string/empty_data"
+        android:textAppearance="?android:attr/textAppearanceLarge"/>
+
+</RelativeLayout>
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_profile.xml]: cuối file nên thêm 1 dòng trống

File: app/src/main/res/layout/activity_home.xml [hunk 32]
```diff
@@ -0,0 +1,26 @@
...
+    <!-- our tablayout to display tabs  -->
+    <android.support.design.widget.TabLayout
+        android:id="@+id/tabLayout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?attr/colorPrimary"
+        android:minHeight="?attr/actionBarSize"
+        android:theme="@style/ThemeOverlay.AppCompat.Dark.ActionBar"/>
+
+</LinearLayout>
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_home.xml]: cuối file nên thêm 1 dòng trống

File: app/src/main/res/layout/activity_favourite.xml [hunk 33]
```diff
@@ -0,0 +1,15 @@
...
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:text="@string/empty_data"
+        android:textAppearance="?android:attr/textAppearanceLarge"/>
+
+</RelativeLayout>
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_favourite.xml]: cuối file nên thêm 1 dòng trống

File: app/src/main/res/layout/fragment_home.xml [hunk 34]
```diff
@@ -0,0 +1,15 @@
...
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:text="@string/empty_data"
+        android:textAppearance="?android:attr/textAppearanceLarge"/>
+
+</RelativeLayout>
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_home.xml]: cuối file nên thêm 1 dòng trống

File: app/src/main/res/values/config.xml [hunk 35]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="tmdb_api_key" templateMergeStrategy="preserve">c2c3be48986f7322bf0ca19e48af1820</string>
+</resources>
```
Comment (daolq3012) [inline app/src/main/res/values/config.xml]: cuối file nên thêm 1 dòng trống

File: app/src/main/res/values/styles.xml [hunk 36]
```diff
@@ -1,11 +1,12 @@
...
+    <!-- changing it to no actionbar -->
+    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
         <!-- Customize your theme here. -->
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
-</resources>
+</resources>
```
Comment (daolq3012) [inline app/src/main/res/values/styles.xml]: cuối file nên thêm 1 dòng trống

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 37]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_home);
+        initViews();
+        handleEvents();
+    }
+
+    private void initViews(){
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java [hunk 38]
```diff
@@ -0,0 +1,62 @@
...
+        mTabLayout.addTab(mTabLayout.newTab().setText(R.string.title_tab_favourite));
+        mTabLayout.addTab(mTabLayout.newTab().setText(R.string.title_tab_profile));
+        mTabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
+        mTabLayout.getTabAt(0).setIcon(tabIcons[0]);
+        mTabLayout.getTabAt(1).setIcon(tabIcons[1]);
+        mTabLayout.getTabAt(2).setIcon(tabIcons[2]);
+        mTabLayout.getTabAt(3).setIcon(tabIcons[3]);
+    }
+
+    private void handleEvents(){
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/androidtrainning/screen/home/MainActivity.java]: ERROR: '{' is not preceded with whitespace.

**Pull request merged**